doc/source/custom_directives.py

```diff
@@ -1252,6 +1255,45 @@ def pregenerate_example_rsts(
             )
 
 
+ray_prefix = "ray-"
+min_version = "1.11.0"
+
+
+def generate_version_url(version):
+    return f"https://docs.ray.io/en/{version}/"
+
+
+#
+# Gets the releases from git, sorts them in semver order,
+# and generates the JSON needed for the version switcher
+#
+def generate_versions_json():
+    try:
+        versions = []
+        tags = subprocess.check_output(["git", "tag"]).decode("utf-8").split()
+        for tag in tags:
+            if ray_prefix in tag:
+                version = tag.split(ray_prefix)[1]
+                if Version(version) >= Version(min_version):
+                    versions.append(version)
+        versions.sort(key=Version, reverse=True)
+
+        version_json_data = [
+            {"version": v, "url": generate_version_url(v)} for v in ["latest", "master"]
+        ] + [
+            {"version": f"releases/{v}", "url": generate_version_url(f"releases-{v}")}
+            for v in versions
+        ]
+
+        with open("versions.json", "w") as f:
+            json.dump(version_json_data, f, indent=4)
+
+        return []
+    except Exception as e:
+        print(f"Error fetching versions: {e}")
```

Comment:
I'm wondering if we should just be passing through without issue if we can't generate the JSON needed for the version picker. What happens if there _is_ an error - does the build process complete without a `versions.json`? What happens if we _don't_ wrap in a try/except here - what kind of errors do we get? It might be more informative to catch a specific error rather than any error.

In any case, I think we should use the logging system if we are going to emit a message to the user, rather than a print.